openmetadata-service/src/main/java/org/openmetadata/service/resources/teams/UserResource.java

```diff
@@ -1040,6 +1055,101 @@ public Response refreshToken(
     return Response.status(Response.Status.OK).entity(authHandler.getNewAccessToken(refreshRequest)).build();
   }
 
+  @GET
+  @Path("/token/personal")
+  @Operation(
+      operationId = "getPersonalAccessToken",
+      summary = "Get personal access token to User",
+      tags = "users",
+      description = "Get a personal access token",
+      responses = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "The user ",
+            content =
+                @Content(mediaType = "application/json", schema = @Schema(implementation = PersonalAccessToken.class))),
+        @ApiResponse(responseCode = "400", description = "Bad request")
+      })
+  public Response getPersonalAccessToken(@Context UriInfo uriInfo, @Context SecurityContext securityContext)
+      throws IOException {
+    String userName = securityContext.getUserPrincipal().getName();
+    User user = dao.getByName(null, userName, getFields("id"), Include.NON_DELETED);
+    List<TokenInterface> tokens =
+        tokenRepository.findByUserIdAndType(user.getId().toString(), TokenType.PERSONAL_ACCESS.value());
+    return Response.status(Response.Status.OK).entity(tokens).build();
+  }
+
+  @PUT
+  @Path("/token/revoke")
+  @Operation(
+      operationId = "revokePersonalAccessToken",
+      summary = "Revoke personal access token to User",
+      tags = "users",
+      description = "Revoke personal access token",
+      responses = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "The user ",
+            content =
+                @Content(mediaType = "application/json", schema = @Schema(implementation = PersonalAccessToken.class))),
+        @ApiResponse(responseCode = "400", description = "Bad request")
+      })
+  public Response revokePersonalAccessToken(@Context UriInfo uriInfo, @Context SecurityContext securityContext)
+      throws IOException {
+    String userName = securityContext.getUserPrincipal().getName();
```

Comment:
we should allow admin to revoke a user's token

Reply:
updated, admin can view and revoke the users personal access token